contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,370 @@
+        self._width_to = kwargs.get("width_to")
+        self._height_to = kwargs.get("height_to")
+        self._framerate = kwargs.get("framerate")
+        # Default extension is mp4, overwrite it if you need
+        self._output_file_extension = "mp4"
+        # Default format of name of artifact
+        self._output_file_name = "resize_{}_{}x{}_to_{}x{}_{}fps.{}".format(
+            self._codec,
+            self._width_from,
+            self._height_from,
+            self._width_to,
+            self._height_to,
+            self._framerate,
+            self._output_file_extension,
+        )
+        # Get the golden sample.
+        # This sample video file will be consumed by any gstreamer piple as
+        # input video.
+        self._golden_sample = get_golden_sample(
+            width=self._width_from,
+            height=self._height_from,
+            framerate=self._framerate,
+        )
+        # A file be treated as the reference file while doing PSNR
+        # Comparision. If you want to use other file as reference while doing
+        # PSNR comparison, please reassign the full path to it
+        self._psnr_reference_file = get_golden_sample(
+            width=self._width_to,
+            height=self._height_to,
+            framerate=self._framerate,
+        )
+        self._output_file_full_path = os.path.join(
+            PLAINBOX_SESSION_SHARE,
+            self._output_file_name,
+        )
+
+    @abstractmethod
+    def _build_command(self) -> str:
+        """
+        Execute the command associated with the given method name for a
+        platform.
+        Returns:
+            str: The corresponding command or an error message if the method
+            is not found.
+        """
+        pass
+
+    def execute_encode_command(self) -> None:
+        logging.debug("Executing Encode Command...")
+        execute_command(self._build_command())
+
+    def compare_psnr(self) -> None:
+        logging.info(
+            "Compare the PSNR: {} vs {}".format(
+                self._psnr_reference_file, self._output_file_full_path
+            )
+        )
+        avg_psnr, _ = get_average_psnr(
+            self._psnr_reference_file, self._output_file_full_path
+        )
+        logging.info("Average PSNR: {}".format(avg_psnr))
+        if avg_psnr < 30 and avg_psnr > 0:
+            raise SystemExit(
+                "Error: The average PSNR value did not reach the acceptable"
+                " threshold (30 dB)"
+            )
+        logging.info("Pass: Average PSNR meets the acceptable threshold")
+
+    def check_metadata(self) -> None:
+        logging.debug("Checking metadata...")
+        outcome = execute_command(
+            cmd="gst-discoverer-1.0 {}".format(self._output_file_full_path)
+        )
+
+        # Check the meta
+        is_metadata_good = True
+        p = self._extract_metadata_property(input=outcome)
+        if not p.get("width") or p.get("width") != self._width_to:
```

Comment:
I think you can directly test `p.get("width") != self._width_to` since `None != str`